src/tools/fuzzing.h

```diff
@@ -2636,6 +2667,154 @@ class TranslateToFuzzReader {
     return builder.makeMemoryFill(dest, value, size);
   }
 
+  Expression* makeTry(Type type) {
+    hangStack.push_back(nullptr);
+    auto* ret = builder.makeTry(makeMaybeBlock(type), makeCatch(type), type);
+    hangStack.pop_back();
+    return ret;
+  }
+
+  Expression* makeCatch(Type type) {
+    // exnref.pop is a pseudo instruction that should follow 'catch'
+    Expression* pop = builder.makePop(Type::exnref);
+
+    // With a small probability, just rethrow it and get done
+    if (oneIn(10)) {
+      return builder.makeRethrow(pop);
+    }
+
+    // Make a block. Note that we don't insert this break in breakableStack:
+    // Because 'exnref.pop' should follow 'catch' and we don't support blocks
+    // taking values yet, if there are instructions that break to this block,
+    // this block cannot be removed when writing binary, which will be a
+    // problem. We also don't make a label for this block.
```

Comment:
Why is it important that the block be removed when writing the binary?

Reply:
Because otherwise it will go between `catch` and the instruction uses `exnref` returned by `catch` (or `exnref.pop` in Binaryen AST).

By the same reason we internally add blocks in `if`'s `then` body and `catch` body, we add blocks in `try` body and `catch` body to handle them easily. Those blocks are not actually printed (both in text and binary) when they don't have a label. So if we make a label for these pseudo-block, we end up making them printed, which is unnecessary for `then`/`else`/`try` bodies but will be a real bug for `catch` body.